pkg/dockerregistry/server/auth.go

```diff
@@ -248,12 +250,18 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				if verifiedPrune {
 					continue
 				}
+				if len(bearerToken) == 0 {
+					return nil, ac.wrapErr(ctx, ErrTokenRequired)
+				}
```

Comment:
why do you not make the verify call, in case anonymous access is sufficient?  (yes i know that is extremely unlikely, but we should be consistent across all api invocations)